.github/workflows/push-master.yaml

```diff
@@ -32,6 +32,6 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPMJS_TOKEN }}
       - name: Report Coverage
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192# pin@1.5.2
```

Comment:
LGTM, just one question, maybe double check are we still using Codecov for the coverage reports?
 
we use Sonarcloud for other new projects, probably we can drop this step at the moment or migrate to Sonarcloud. Also, the v1 version is deprecated as this action said.

Reply:
This workflow generates the test coverage when push code against master branch, kinda unusual.

Reply:
But let's don't expand the scope in this PR, just pin the github actions as required,

Reply:
Yes we've preferred to use codecov for open source repo's over sonar cloud. The reason it occurs against master is to update the badge shown in the root readme of the repo.

Reply:
@tplooker Got it. Thanks for the clarification.